### DISCUSSION TITLE
How HTTP subscribe API works ?? If I subscribe through http API then how emqx broker will send the data to my application if any message comes on subscribed topic

### DISCUSSION THREAD

**Discussion by kumarvijay95** [emqx/emqx#7858] [Q&A] [answered]
Hi All, I am using emqx HTTP APIs to work with emqx broker and I Have created my application in java language. 
Now my question is If I hit subscribe API i.e "POST /api/v4/mqtt/subscribe" then how emqx will send the message on subscribed topic. Please help me on this. It's urgent.

**Comment by zmstone** [comment, marked as answer]
Hi. The subscribe API is to help an MQTT client subscribe to certain topic — in case the client dose not or cannot send subscribe requests.

The http client doesn’t not receive messages.
